zaza/openstack/charm_tests/gnocchi/tests.py

```diff
@@ -89,6 +92,41 @@ def setUpClass(cls):
         # Create AWS compatible application credentials in Keystone
         cls.ec2_creds = ks_client.ec2.create(user_id, project_id)
 
+    def test_upload_external_cert(self):
+        """Verify that the external CA is uploaded correctly."""
+        logging.info('Changing value for trusted-external-ca-cert.')
+        ca_cert_option = 'trusted-external-ca-cert'
+        ppk, cert = utilities.cert.generate_cert('gnocchi_test.ci.local')
+        b64_cert = base64.b64encode(cert).decode()
+        config = {
+            ca_cert_option: b64_cert,
+        }
+        model.set_application_config(
+            'gnocchi',
+            config
+        )
+        model.block_until_all_units_idle()
+
+    def test_validate_cert_location(self):
```

Comment:
We don't want to rely on the order of execution of the tests because we have noticed in the past that the order is not 100% guaranteed. This leads to a flaky gate that fails every now and then and it's quite hard to figure out why.

I guess what you want here is to merge all 3 test functions into one which is covering the entire scenario, but I'm also fine if you have a better idea.

Reply:
Ah I did not know that, I was just striving for smaller functions. Sure, I will put them all together.

Reply:
Update pushed